airflow/www/views.py

```diff
@@ -3574,23 +3574,28 @@ def datasets_summary(self):
 
             count_query = session.query(func.count(DatasetModel.id))
 
+            has_event_filters = bool(updated_before or updated_after)
+
             query = (
                 session.query(
                     DatasetModel.id,
                     DatasetModel.uri,
                     func.max(DatasetEvent.timestamp).label("last_dataset_update"),
-                    func.count(distinct(DatasetEvent.id)).label("total_updates"),
+                    func.count(func.sum(func.case(DatasetEvent.id.is_not(None), 1, else_=0))).label(
+                        "total_updates"
+                    ),
                 )
-                .outerjoin(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id)
+                .join(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id, isouter=not has_event_filters)
                 .group_by(
                     DatasetModel.id,
                     DatasetModel.uri,
                 )
                 .order_by(*order_by)
             )
 
-            if updated_before or updated_after:
-                count_query = count_query.outerjoin(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id)
+            if has_event_filters:
+                count_query = count_query.join(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id)
```

Comment:
if event filters, need to join, and inner is fine.   if not, we don't need to join because it's just counting datasets